src/pyqasm/visitor.py

```diff
@@ -462,6 +462,80 @@ def _get_op_bits(
 
         return openqasm_bits
 
+    def _check_type_size(
+        self, statement: qasm3_ast.Statement, var_name: str, base_type: Any, is_const: bool
+    ) -> int:
+        """Get the size of the given variable type.
+
+        Args:
+            statement: current statement to get span.
+            var_name(str): variable name of the current operation.
+            base_type (Any): Base type of the variable.
+            is_const (bool): whether the statement is constant declaration or not.
+        Returns:
+            Int: size of the variable base type.
+        """
+        base_size = 1
+        if not isinstance(base_type, qasm3_ast.BoolType):
+            initial_size = 1 if isinstance(base_type, qasm3_ast.BitType) else 32
+            try:
+                base_size = (
+                    initial_size
+                    if not hasattr(base_type, "size") or base_type.size is None
+                    else Qasm3ExprEvaluator.evaluate_expression(base_type.size, const_expr=True)[0]
+                )
+                if is_const:
```

Comment:
Shouldn't this be executed in every case irrespective of whether we have const or not? Eg. for `int[-16] a;` this should be invoked as well right?